src/data_structure/economic_structure.jl

```diff
@@ -0,0 +1,491 @@
+#############################################################################
+# Copyright (C) 2017 - 2018  Spine Project
+#
+# This file is part of SpineOpt.
+#
+# SpineOpt is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# SpineOpt is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#############################################################################
+
+
+# API
+"""
+    generate_economic_structure!(m::Model)
+"""
+function generate_economic_structure!(m::Model;log_level=3)
+    for (obj, name) in [(unit,:unit),(node,:storage),(connection,:connection)]
+        @timelog log_level 3 "- [Generated discounted durations for $(obj)s]" generate_discount_timeslice_duration!(m::Model,obj, name)
+    end
+    !isempty([
+        model__default_investment_temporal_block()...,
+        node__investment_temporal_block()...,
+        unit__investment_temporal_block()...,
+        connection__investment_temporal_block()...
+    ]) || return
```

Comment:
This splatting is not very fast, maybe use vcat - or repace the `...,` by just `;`